### PR TITLE
CNF-12622: Add Image Based Install reference templates

### PR DESCRIPTION
**Pull request by sakhoury**
This PR contains the Image-based install reference templates.

### CODE REVIEW

File: internal/templates/image-based-install/template.go [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+	data := make(map[string]string)
+	data["ClusterDeployment"] = ClusterDeployment
+	data["ManagedCluster"] = ManagedCluster
+	data["KlusterletAddonConfig"] = KlusterletAddonConfig
+	return data
+}
+
+func GetNodeTemplates() map[string]string {
+	data := make(map[string]string)
+	data["ImageClusterInstall"] = ImageClusterInstall
```
Comment (eranco74) [inline internal/templates/image-based-install/template.go:174]: I'm not sure what's the meaning of the division between cluster and node templates. 
Although ImageClusterInstall currently (and for the foreseeable future) only supports single node installation I think that the ImageClusterInstall resource belongs to the clusterTemplates.

Comment (sakhoury) [reply internal/templates/image-based-install/template.go:174]: The current `v1alpha1` version of the `ImageClusterInstall` manifest references node-specific fields such as the hostname and BMH references, in addition to network-related configs pertaining to the node. As per the design of the cluster and node templates for the SiteConfig API, a cluster template should only reference cluster (site-wide) fields, whereas, a node template may reference both cluster and node-specific fields. For example, the `AgentClusterInstall` CR only refers to cluster-wide fields, thus, it is categorized as a cluster template.

**Review by eranco74 — APPROVED** [review]

**Pull request merged**